multiversx_sdk_cli/delegation/staking_provider.py

```diff
@@ -304,27 +305,21 @@ def _parse_public_bls_keys(self, public_bls_keys: str) -> List[ValidatorPublicKe
     def _get_public_keys_and_signed_messages(self, args: Any) -> Tuple[List[ValidatorPublicKey], List[bytes]]:
         validators_file_path = Path(args.validators_file).expanduser()
         validators_file = ValidatorsFile(validators_file_path)
+        signers = validators_file.load_signers()
 
-        pubkey = self._get_pubkey_for_signing(args)
+        pubkey = self._get_pubkey_to_be_signed(args)
 
         public_keys: List[ValidatorPublicKey] = []
         signed_messages: List[bytes] = []
-        for validator in validators_file.get_validators_list():
-            # Get path of "pemFile", make it absolute
-            validator_pem = Path(validator.get("pemFile")).expanduser()
-            validator_pem = validator_pem if validator_pem.is_absolute() else validators_file_path.parent / validator_pem
+        for signer in signers:
+            signed_message = signer.sign(pubkey)
 
-            pem_file = ValidatorPEM.from_file(validator_pem)
-
-            validator_signer = ValidatorSigner(pem_file.secret_key)
-            signed_message = validator_signer.sign(pubkey)
-
-            public_keys.append(pem_file.secret_key.generate_public_key())
+            public_keys.append(signer.secret_key.generate_public_key())
             signed_messages.append(signed_message)
 
         return public_keys, signed_messages
 
-    def _get_pubkey_for_signing(self, args: Any) -> bytes:
+    def _get_pubkey_to_be_signed(self, args: Any) -> bytes:
```

Comment:
Now I see, this does not work with Ledger yet.

Reply:
Should work now.